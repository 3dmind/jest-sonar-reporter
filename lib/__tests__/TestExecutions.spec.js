'use strict'

const xml = require('xml')
const TestExecutions = require('../TestExecutions')

describe('TestExecutions', () => {
  test('root: <testExecutions version="1">', () => {
    const mock = {testResults: []}
    const testExecutions = new TestExecutions(mock)

    const actualReport = xml(testExecutions)

    expect(actualReport).toMatchSnapshot()
  })

  test('file tag', () => {
    const mock = {
      testResults: [
        {
          testFilePath: 'test/FooTest.js',
          testResults: []
        },
        {
          testFilePath: 'test/BarTest.js',
          testResults: []
        }
      ]
    }
    const testExecutions = new TestExecutions(mock)

    const actualReport = xml(testExecutions, true)

    expect(actualReport).toMatchSnapshot()
  })

  test('full report', () => {
    const mock = {
      testResults: [
        {
          testFilePath: 'test/FooTest.js',
          testResults: [
            {
              title: 'lorem ipsum'
            }
          ]
        },
        {
          testFilePath: 'test/BarTest.js',
          testResults: [
            {
              title: 'lorem ipsum',
              status: 'failed',
              failureMessages: ['Lorem ipsum']
            }
          ]
        }
      ]
    }
    const testExecutions = new TestExecutions(mock)

    const actualReport = xml(testExecutions, true)

    expect(actualReport).toMatchSnapshot()
  })
})
