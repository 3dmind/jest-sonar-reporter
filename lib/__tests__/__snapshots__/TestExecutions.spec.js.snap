// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestExecutions file tag 1`] = `
"<testExecutions version=\\"1\\">
    <file path=\\"test/FooTest.js\\">
    </file>
    <file path=\\"test/BarTest.js\\">
    </file>
</testExecutions>"
`;

exports[`TestExecutions full report 1`] = `
"<testExecutions version=\\"1\\">
    <file path=\\"test/FooTest.js\\">
        <testCase name=\\"lorem ipsum\\" duration=\\"0\\"/>
    </file>
    <file path=\\"test/BarTest.js\\">
        <testCase name=\\"lorem ipsum\\" duration=\\"0\\">
            <failure message=\\"Lorem ipsum\\"/>
        </testCase>
    </file>
</testExecutions>"
`;

exports[`TestExecutions root: <testExecutions version="1"> 1`] = `"<testExecutions version=\\"1\\"></testExecutions>"`;
